src/panels/lovelace/cards/hui-thermostat-card.ts

```diff
@@ -0,0 +1,628 @@
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+
+const thermostatConfig = {
+  radius: 150,
+  min: 50,
+  max: 90,
+  step: 1,
+  circleShape: "pie",
+  sliderType: "min-range",
+  value: 70,
+  startAngle: 315,
+  width: 5,
+  lineCap: "round",
+  handleSize: "+10",
+  handleShape: "dot",
+  showTooltip: true,
+};
+
+const modeColor = {
+  auto: "green",
+  heat: "#FF8100",
+  cool: "#2b9af9",
+  off: "#8a8a8a",
+};
+
+const modeIcons = {
+  auto: "mdi:autorenew",
+  heat: "hass:fire",
+  cool: "hass:snowflake",
+  off: "mdi:fan-off",
+};
+
+const thermostatState = {
+  entity_id: "climate.thermostat",
+  state: "cool",
+  attributes: {
+    current_temperature: 73,
+    min_temp: 50,
+    max_temp: 90,
+    temperature: 70,
+    target_temp_high: 75,
+    target_temp_low: 65,
+    fan_mode: "Auto Low",
+    fan_list: ["On Low", "On High", "Auto Low", "Auto High", "Off"],
+    operation_mode: "cool",
+    operation_list: ["heat", "cool", "auto", "off"],
+    hold_mode: "home",
+    swing_mode: "Auto",
+    swing_list: ["Auto", "1", "2", "3", "Off"],
+    unit_of_measurement: "°F",
+    friendly_name: "Ecobee",
+    supported_features: 1014,
+  },
+  last_changed: "2018-07-19T10:44:46.200333+00:00",
+  last_updated: "2018-07-19T10:44:46.200333+00:00",
+};
+
+interface Config extends LovelaceConfig {
+  entity: string;
+}
+
+export class HuiThermostatCard extends HassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  protected hass?: HomeAssistant;
+  protected config?: Config;
+
+  static get properties() {
+    return {
+      hass: {},
+      config: {},
+    };
+  }
+
+  public getCardSize() {
+    return 4;
+  }
+
+  public setConfig(config: Config) {
+    if (!config.entity || config.entity.split(".")[0] !== "climate") {
+      throw new Error("Specify an entity from within the climate domain.");
+    }
+
+    this.config = config;
+  }
+
+  protected render() {
+    const thermostat = { "climate.thermostat": thermostatState };
+    this.hass!.states = { ...thermostat, ...this.hass!.states };
+    const stateObj = this.hass!.states[this.config!.entity];
+
+    return html`
+      ${this.renderStyle()}
+      <ha-card>
+        <div
+          .hass="${this.hass}"
+          .config="${this.config}"
+          .localize="${this.localize}"
+          id="thermostat"
+        ></div>
+        <div class="climate-info">
+          <div id="set-temperature">${stateObj.attributes.temperature}</div>
+          <div class="current-mode">${this.localize(
+            `state.climate.${stateObj.state}`
+          )}ing</div>
+          <div class="modes">
+            <ha-icon
+              class="${
+                stateObj.attributes.operation_mode === "auto"
+                  ? "selected-icon"
+                  : ""
+              }"
+              entity="${stateObj.entity_id}"
+              id="auto"
+              @click="${this.handleModeClick}"
```

Comment:
I think that we should stick with keeping event listeners be prefixed with `_` if private. That way custom UI will know it's not for usage.

Reply:
Good point